### ISSUE TITLE
Loosing focus while typing in Ferdi >= 5.4.1-beta.5

### ISSUE THREAD

**Issue by VOOM108** [getferdi/ferdi#343]
**Describe the bug**
This started with the latest 5.4.1 beta.5 (I skipped beta.6) and continues with the current 5.4.3: while typing in any service, or at least any chat like Slack, WhatsApp, Messenger etc., Ferdi looses focus and I have to click into the text input field again to continue typing. 

The frequency of this happening is irregular, but I can hardly finish any message without it happening at least once. 

**Desktop (please complete the following information):**

- OS: Win10
- Version 5.4.3

(Have not tried 5.4.4 beta.1 yet)

**Comment by kimmminjae** [comment]
Same thing happening to me

OS: Windows 10 Enterprise 1909
Version 5.4.1 beta-4

**Comment by ccioanca** [comment]
Same thing's happening here. 

Though sometimes it seems like Microsoft Teams is the problem. If an input box loses focus somewhere, I can usually find the jumbled remains of text in Microsoft Teams for some reason. 

Not sure if this is a clue, or if my experience is a one-off.

**Comment by VOOM108** [comment]
I don't even know, what Microsoft Teams is ;) My lost input usually goes nowhere at all. 

I suppose, where the focus goes is rather random, at the center is Ferdi loosing focus. Because if certain apps stole the focus, other apps would loose focus, too, which is not the case.

**Comment by VOOM108** [comment]
Any ideas yet anyone? This makes using Ferdi virtually impossible. I will have to go back to an older Version 5.4.1 beta-3 or so that had other problems until a solution is found :|

**Comment by kimmminjae** [comment]
Yes I need this fixed! This problem happened just now as I was typing this.

**Comment by vantezzen** [comment]
This seems to be a Windows-only problem as I can't reproduce it on macOS.

**Comment by VOOM108** [comment]
Maybe this helps to find sth: I went back to the last version I remember to work flawlessly for me concerning this: 5.4.1 beta-4 The issue persists. I am pretty sure this occured first - for me - on beta-5

**Comment by VOOM108** [comment]
> 
> 
> Same thing happening to me
> 
> OS: Windows 10 Enterprise 1909
> Version 5.4.1 beta-4

While this user refers to beta-4 as well

**Comment by vantezzen** [comment]
Could you please provide me your Ferdi's debug information? In the menubar under "Help" > "Copy Debug Information"

**Comment by VOOM108** [comment]
Since downgrading did not help back at 5.4.4-b1

[debug-info-5.4.4-b1.txt](https://github.com/getferdi/ferdi/files/4164676/debug-info-5.4.4-b1.txt)

**Comment by VOOM108** [comment]
Does anyone of the affected use DisplayFusion? (Just an idea, though I made no recent changes to the DF-settings before this started happening)

**Comment by kimmminjae** [comment]
> Does anyone of the affected use DisplayFusion? (Just an idea, though I made no recent changes to the DF-settings before this started happening)

Nope. I don't use DisplayFusion

**Comment by VOOM108** [comment]
I did some more testing with two focus-logging tools, namely happydroid focus and "Windows Focus Logger" by Adminscope

The result is, that both note no other process or app stealing focus from ferdi. The register nothing, while typing is blocked and the focus is taken out of the respective input form of whatever service within ferdi. So that looks like some internal process of ferdi itself takes the focus out of the input field randomly. 

I will start investigating, if some service could cause this. Remove the least used ones and maybe change the settings to have some services paused. 

The issue still is a big fat nuisance while using ferdi for me :|

**Comment by kytwb** [comment]
Could it be related to the updater being broken? Sometimes I notice Ferdi jumping in my macOS task bar when not focused, maybe that make loose focus when focused? without prompting me of an udpate (on an old install; the updater is broken there).

**Comment by justinebateman** [comment]
For what it's worth, I have been experiencing this too in Slack. I am on Mac Catalina with Ferdi version 5.4.3 (5.4.3.994)

**Comment by mahadevans87** [comment]
@justinebateman Can you help me reproduce this? I am using 5.4.3 (WIndows 10) and can't seem to reproduce this.

**Comment by ccioanca** [comment]
Just an update from me: 

Seems like the context menu for text correction works. but the normal context menu for copy/paste/etc doesn't. It flickers and disappears just like others are reporting. Not sure if somehow the correction menu is taking over or something. Just a hail mary here.

**Comment by justinebateman** [comment]
I can't say for sure what the problem is, but I can back up @ccioanca 's statement that 

> "Though sometimes it seems like Microsoft Teams is the problem. If an input box loses focus somewhere, I can usually find the jumbled remains of text in Microsoft Teams for some reason."

The exact same thing was happening to me, I was typing in slack, the textbox loses focus, I switch to Teams and I can see the remainder of what I was typing in the Teams chatbox. 

I removed Teams from my Ferdi completely and I haven't had this happen to me since. I currently have 4 Slack workspaces, and 1 Whatsapp workspace on my Ferdi with no issue.

Others have stated that they don't have Teams in their Ferdi and this is happening to them, so I think the issue is more around a different app "stealing" focus from Slack

**Comment by basalphenaar** [comment]
This issue started happening for me after updating to 5.5.0-gm. Reverting back to 5.4.4-beta.3 fixed it for me.

**Comment by mahadevans87** [comment]
@vantezzen I was looking at the latest develop version and could finally manage to reproduce the error - I noticed that in the debug console, as we type in some text field, the recipe webview's initialize function is getting called repeatedly which I suspect could be the reason for losing focus -

![image](https://user-images.githubusercontent.com/1255523/79689633-c4091180-8273-11ea-943b-e62afbac8704.png)

**Comment by mahadevans87** [comment]
Was look at the whatsapp recipe and found the following piece of code - 
```

setTimeout(() => {
  const elem = document.querySelector('.landing-title.version-title');

  if (elem && elem.innerText.toLowerCase().includes('google chrome')) {
    window.location.reload();
  }
}, 1000);

Could this be the culprit?
```

**Comment by kytwb** [comment]
@mahadevans87 I think we can/should 🙏

**Comment by mahadevans87** [comment]
@kytwb There are atleast 7 recipes which do a setTimeout and do some kind of work repeatedly and that includes Whatsapp, slack, rocket chat, reddit, android messages, facebook pages. I suspect people reporting the issue have atleast one of these services running in Ferdi. It would be good if someone can confirm.

**Comment by basalphenaar** [comment]
> @kytwb There are atleast 7 recipes which do a setTimeout and do some kind of work repeatedly and that includes Whatsapp, slack, rocket chat, reddit, android messages, facebook pages. I suspect people reporting the issue have atleast one of these services running in Ferdi. It would be good if someone can confirm.

Yes, I have WhatsApp and multiple Slack instances running.

**Comment by justinebateman** [comment]
I can also confirm - whatsapp and multiple slack instances running. Sounds like you are onto something, fingers crossed this resolves it 🤞

**Comment by mahadevans87** [comment]
That is reassuring. I will try to fix this and raise a PR later today.

**Comment by mahadevans87** [comment]
> 
> 
> I can also confirm - whatsapp and multiple slack instances running. Sounds like you are onto something, fingers crossed this resolves it 🤞

@justinebateman @basalphenaar Can you confirm if your slack service is set in hibernation mode? And you are losing focus whenever you try to reload slack and start typing?

**Comment by justinebateman** [comment]
I don't believe I have hibernation mode switched on

![image](https://user-images.githubusercontent.com/28674290/79753457-df285f80-830d-11ea-9644-c4620ab38386.png)

**Comment by mahadevans87** [comment]
> I don't believe I have hibernation mode switched on

ok thanks for that. Can you provide a list of all the services that you are using? This will help me figure out the problem easier and I can focus on those recipes

**Comment by VOOM108** [comment]
I can also confirm using WhatsApp and Slack. I don't use hibernation. As mentioned above, changing some settings made the effect disappear, though I am not sure what exactly did the trick. Still using WhatsApp and one Slack instance

**Comment by justinebateman** [comment]
I'm using four Slack workspaces and one Whatsapp instance. I did have a Microsoft Teams instance and the issue was worse when I had it, I removed Teams from Ferdi and thought the issue was resolved but I've had it happen once or twice since.

**Comment by kytwb** [comment]
@justinebateman maybe the debug info available via the _Help_ menu could be useful?

<img width="248" alt="Screen Shot 2020-04-20 at 12 56 39 PM" src="https://user-images.githubusercontent.com/412895/79754043-6a055c00-8306-11ea-8629-8f406b9fe979.png">

**Comment by justinebateman** [comment]
[justine-ferdi-debug.txt](https://github.com/getferdi/ferdi/files/4503669/justine-ferdi-debug.txt)

Here's mine, I've tried to remove any "sensitive" info, hopefully it helps

**Comment by mahadevans87** [comment]
> 
> 
> [justine-ferdi-debug.txt](https://github.com/getferdi/ferdi/files/4503669/justine-ferdi-debug.txt)
> 
> Here's mine, I've tried to remove any "sensitive" info, hopefully it helps

I see that you have got a dev-recipe called **notion**. Any idea if I can see the source code of the recipe?

**Comment by kytwb** [comment]
Fix submitted by @mahadevans87 is available in [v5.5.0-gm.2](https://github.com/getferdi/ferdi/releases/tag/v5.5.0-gm.2).  
Please let us know if this is still happening with this version, in which case we will re-open 🙏

**Comment by basalphenaar** [comment]
Thanks, issue has been fixed for me in v5.5.0-gm.2!
